### ISSUE TITLE
Error duplicate 'row.names'

### ISSUE THREAD

**Issue by Ryo-1234** [satijalab/seurat#1710]
Hi 

When I try to read a CSV file in GEO database sample using row.names = 1, I get this error:
```
library(Seurat)
> library(ggplot2)
> cancer.rna <- read.csv(file = "/Users/ryo/desktop/GSE84133/GSM2230757_human1_umifm_countsnew14.csv", sep = ",",header = TRUE, row.names = 1)
Error in read.table(file = file, header = header, sep = sep, quote = quote,  : 
  duplicate 'row.names' are not allowed
> dim(cancer.rna)
Error: object 'cancer.rna' not found
```

and When I try to read CSV file  in my sample  using row.names = NULL, It did work. But I get warning message about CreateSeuratObject；
```
> library(Seurat)
> library(ggplot2)
> cancer.rna <- read.csv(file = "/Users/ryo/desktop/GSE84133/GSM2230757_human1_umifm_countsnew14.csv", sep = ",",header = TRUE, row.names = NULL)
> dim(cancer.rna)
[1] 16342  1651
> cancer.rna[1:5, 1:5]
       X CTGTTTACACCGCTAG CTCTACGGTGTGGCTC AGCAGCCAGGCTCATT GAATAAGAGATCCCAT
1   A1BG                0                0                0                0
2   A1CF                0                0                0                2
3    A2M                4                0                0                0
4  A2ML1                0                0                0                0
5 A4GALT                0                0                0                0
> cancer <- CreateSeuratObject(counts = cancer.rna) 
**Warning message:
In storage.mode(from) <- "double" : NAs introduced by coercion**
> cancer <- NormalizeData(cancer)
Performing log-normalization
0%   10   20   30   40   50   60   70   80   90   100%
[----|----|----|----|----|----|----|----|----|----|
**************************************************|
Error: Attempting to add a different number of cells and/or features
```

I think "row.names = 1" is the best one in this case, because Seurat tutorial was used "row.names = 1". Could you tell me how to improve to the error message or Warning message ?

**Comment by yuhanH** [comment]
From the `read.csv` error, it looks that you have duplicated gene names in your counts file. 
You could rename the duplicated genes or delete the duplicated one.  
You could `read.csv` file with `row.names=NULL`, and test `which(duplicated(cancer.rna$X))`

**Comment by diazdc** [comment]
In many cases gene names are repeated because there isn't consensus over which coding sequence represents the common name. I would exercise a bit of caution if you decide throw away genes that have duplicated names, because doing so may affect how your cells cluster, especially if you remove a gene that has a highly variable expression pattern. 

To that end, you can use the R function make.unique() to.. you guessed it.. make your gene names unique. This works by appending a number with a period delimiter for every repeat name encountered. Seurat actually uses this method in its Read10X function by default. 

To summarize:
```
cancer.rna <- read.csv(file = "/Users/ryo/desktop/GSE84133/GSM2230757_human1_umifm_countsnew14.csv", sep = ",",header = TRUE, row.names = NULL)
names <- make.unique(cancer.rna$X)
rownames(cancer.rna) <- names
cancer.rna <- cancer.rna[,-1] # get rid of old names
cancer <- CreateSeuratObject(counts = cancer.rna)
```

**Comment by yuhanH** [comment]
Totally agree.  I think the code from diazdc could solve your issue.

**Comment by Ryo-1234** [comment]
Many thanks, diazdc and yuhanH !  I think so too. It is very important that row name change to the unique name.
I read the following but received an error.
Could you tell me how to improve the error message or Warning message?

```
> cancer.rna <- read.csv(file = "/Users/ryo/desktop/GSE84133/GSM2230757_human1_umifm_countsnew14.csv", sep = ",",header = TRUE, row.names = NULL)
> dim(cancer.rna)
[1] 16342  1651
> cancer.rna[1:5, 1:5]
       X CTGTTTACACCGCTAG CTCTACGGTGTGGCTC AGCAGCCAGGCTCATT GAATAAGAGATCCCAT
1   A1BG                0                0                0                0
2   A1CF                0                0                0                2
3    A2M                4                0                0                0
4  A2ML1                0                0                0                0
5 A4GALT                0                0                0                0
> names<- make.unique(cancer.rna$X)
Error in make.unique(cancer.rna$X) : 'names' must be a character vector
```

**Comment by yuhanH** [comment]
I think those genes are imported as factor. 
`names<- make.unique(as.character(cancer.rna$X))`
This can solve your issue.

**Comment by Ryo-1234** [comment]
Woah, that's cool! It does work and then, I could made a t-SNE plot. So, Many thanks yuhanH.
